### PR TITLE
Feature/add new habit

### PR DESCRIPTION
**Pull request by GregVaz**
## Explain why this change is being made
As an user, I want to add new habits so that I can create one habit

## Explain how this is accomplished

- One habit is created 
- Once the habit is created it redirects to you to the habit details page

## How do you manually test this?
1. Visit the habits page
2. Click the button Create new Habit
3. Fill in the form with the name, description, start date and habit's duration
4. Click button Create Habit
5. It will show you the information of the habit 

#### Screenshots (if any)

### CODE REVIEW

**Review by LoginALee — APPROVED** [review]

File: spec/features/create_new_habit_spec.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  scenario "create a new habit" do
+    visit habits_path
+    click_button 'Add New Habit'
+    expect(page).to have_content 'Create a new habit'
+
+    fill_in 'Name', with: habit[:name]
+    fill_in 'Description', with: habit[:description]
+    fill_in 'Start date', with: habit[:start_date]
+    fill_in 'Quantity', with: 30
```
Comment (ese-varo) [inline spec/features/create_new_habit_spec.rb]: is it always going to have 30 days as quantity? if not, what do you think about random numbers between a range?

File: spec/features/create_new_habit_spec.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  scenario "create a new habit with an invalid name", js: true do
+    visit habits_path
+    click_button 'Add New Habit'
+    expect(page).to have_content 'Create a new habit'
+
+    fill_in 'Name', with: ''
+    fill_in 'Description', with: habit[:description]
+    fill_in 'Start date', with: habit[:start_date]
+    fill_in 'Quantity', with: 30
```
Comment (ese-varo) [inline spec/features/create_new_habit_spec.rb]: same here

**Review by ese-varo — APPROVED** [review]
I left some comments, but it actually seems great, well done guys! :rocket:

File: spec/features/create_new_habit_spec.rb (hunk 1 above)
Comment (GregVaz) [reply spec/features/create_new_habit_spec.rb]: Done, that's a good point, yeah it could be with the range between 21 and 66

**Review by LoginALee — APPROVED** [review]
Cool, Good job guys :heart:

**Review by ese-varo — APPROVED** [review]

File: app/controllers/habits_controller.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  def show
+  end
+
+  def set_habit
+    @habit = Habit.find(params[:id])
+  end
+
+  def end_date_param
+    params[:habit][:start_date].to_date + params[:habit][:quantity].to_i.days
```
Comment (OmauryMarmolejo) [inline app/controllers/habits_controller.rb]: What happens if the start_date is not present? Also, What is `quantity`?

File: app/controllers/habits_controller.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  def habit_params
+    params.require(:habit).permit(:name, :description, :start_date).merge(:end_date => end_date_param)
```
Comment (OmauryMarmolejo) [inline app/controllers/habits_controller.rb]: Please, try to use colon syntax `end_date: ...`

File: app/helpers/habits_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module HabitsHelper
```
Comment (OmauryMarmolejo) [inline app/helpers/habits_helper.rb]: Is this file needed?

File: app/views/habits/index.html.haml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+%button.btn.btn-primary{ 'type': "button", 'data-bs-toggle': "modal", 'data-bs-target': "#staticBackdrop"} Add New Habit
+
+.modal.fade#staticBackdrop{ 'data-bs-backdrop': "static", 'data-bs-keyboard':"false", 'tabindex': "-1", 'aria-labelledby': "staticBackdropLabel", 'aria-hidden': "true" }
+  .modal-dialog
+    .modal-content
+      .modal-header
+        %h5.modal-title#staticBackdropLabel>Create a new habit
```
Comment (OmauryMarmolejo) [inline app/views/habits/index.html.haml]: Is this line correct?

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -11,6 +11,8 @@
   </head>
 
   <body>
-    <%= yield %>
+    <div class='container'>
```
Comment (OmauryMarmolejo) [inline app/views/layouts/application.html.erb]: You include this div for every place you use the application layout. There are no exceptions?

**Review by OmauryMarmolejo — CHANGES_REQUESTED** [review]

File: app/views/layouts/application.html.erb (hunk 7 above)
Comment (jvargas98) [reply app/views/layouts/application.html.erb]: yes, we believe that there are no exceptions and that it cannot affect future features. 🤔

File: app/views/habits/index.html.haml (hunk 6 above)
Comment (jvargas98) [reply app/views/habits/index.html.haml]: already changed, thanks

File: app/controllers/habits_controller.rb (hunk 3 above)
Comment (GregVaz) [reply app/controllers/habits_controller.rb]: We already solved this, we change the name of quantity by habit_duration, it is more understandable
And raise an exception when the attribute in params does not exist

File: app/controllers/habits_controller.rb (hunk 4 above)
Comment (GregVaz) [reply app/controllers/habits_controller.rb]: Sure, thanks :D

File: app/helpers/habits_helper.rb (hunk 5 above)
Comment (jvargas98) [reply app/helpers/habits_helper.rb]: we have already removed it

**Review by ese-varo — APPROVED** [review]

**Review by LoginALee — APPROVED** [review]

**Review by OmauryMarmolejo — APPROVED** [review]

**Pull request closed without merging**